model.py

```diff
@@ -9021,13 +9047,71 @@ class Collection(Base):
 
     def __repr__(self):
         return (u'<Collection "%s"/"%s" ID=%d>' %
-                (self.name, self.protocol, self.id)).encode('utf8')
+                (self.name, self.protocol, self.id)).encode('utf8')        
+
+    @classmethod
+    def by_name_and_protocol(cls, _db, name, protocol):
```

Comment:
Looking up and creating Collections has become more complicated because what used to be a field of Collection is now in an associated object.